bip-0352.mediawiki

```diff
@@ -0,0 +1,472 @@
+* Let ''a = a<sub>0</sub> + a<sub>1</sub> + ... + a<sub>n</sub>'', where each ''a<sub>i</sub>'' has been negated if necessary
+* Group receiver silent payment addresses by ''B<sub>scan</sub>'' (e.g. each group consists of one ''B<sub>scan</sub>'' and one or more ''B<sub>m</sub>'')
+* For each group:
+** Let ''ecdh_shared_secret = outpoints_hash·a·B<sub>scan</sub>''
+** Let ''k = 0''
+** For each ''B<sub>m</sub>'' in the group:
+*** Let ''t<sub>k</sub> = sha256(ser<sub>P</sub>(ecdh_shared_secret) || ser<sub>32</sub>(k))''
+**** If ''t<sub>k</sub>'' = 0 or ''t<sub>k</sub>'' ≥ ''n'', fail
+*** Let ''P<sub>mn</sub> = B<sub>m</sub> + t<sub>k</sub>·G''
+*** Encode ''P<sub>mn</sub>'' as a [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP341] taproot output
+*** Optionally, repeat with k++ to create additional outputs for the current ''B<sub>m</sub>''
+*** If no additional outputs are required, continue to the next ''B<sub>m</sub>'' with ''k++''<ref name="why_not_the_same_tn">''' Why not re-use ''t<sub>k</sub>'' when paying different labels to the same receiver?''' If paying the same entity but to two separate labeled addresses in the same transaction without incrementing ''k'', an outside observer could subtract the two output values and observe that this value is the same as the difference between two published silent payment addresses and learn who the recipient is.</ref>
+** Optionally, if the sending wallet implements receiving silent payments, it can create change outputs by sending to its own silent payment address using label ''m = 0'', following the steps above
+
+=== Receiver ===
+
+==== Key Derivation ====
+
+Two keys are needed to create a silent payments address: the spend key and the scan key. Wallet software SHOULD use BIP32 hardened derivation<ref name="bip32_derivation">'''Why use BIP32 hardened derivation?''' Using BIP32 derivation allows users to add silent payments to an existing master seed. It also ensures that a user's silent payment funds are recoverable in any BIP32/BIP43 compatible wallet. Using hardened derivation ensures that it is safe to export the scan private key without exposing the master key or spend private key.</ref> for the spend key and hash it to obtain the scan key.
```

Comment:
> Wallet software SHOULD use BIP32 hardened derivation for the spend key

It's actually more important that the scan key (which might be given to a 3rd party scanning service or online computer) use hardened derivation. The spend key should never leave the signing device, so it matters less.

> and hash it to obtain the scan key.

This seems to conflict with the following line that specifies separate derivations for the two keys.

Reply:
> This seems to conflict with the following line that specifies separate derivations for the two keys.

Indeed. In any case, I think the spec here should give the exact derivation method, and then the following method using BIP32 seems nice.

Reply:
This is another thing we've gone back and forth on, and ultimately decided on specifying BIP32 hardened derivation for the scan key, which is essential (as @brandonblack notes) so that the scan key can safely be exported to a 3rd party. I'll update the wording to be consistent here.